### PR TITLE
[FLINK-26611] Document operator config options

### PR DESCRIPTION
**Pull request by bgeng777**
- Add `Deployment` -> `Configuration` tab in the doc website. This page can be used to document other configs as well. (e.g. https://issues.apache.org/jira/browse/FLINK-27008)
- Add document of operator config options.

### CODE REVIEW

File: flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/config/OperatorConfigOptions.java [hunk 1]
```diff
@@ -33,7 +33,7 @@
                     .durationType()
                     .defaultValue(Duration.ofSeconds(60))
                     .withDescription(
-                            "The interval for the controller to reschedule the reconcile process");
+                            "The interval for the controller to reschedule the reconcile process.");
```
Comment (SteNicholas) [inline flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/config/OperatorConfigOptions.java:36]: I have updated this in #141.

File: flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/config/OperatorConfigOptions.java [hunk 2]
```diff
@@ -61,7 +61,7 @@
                     .durationType()
                     .defaultValue(Duration.ofSeconds(10))
                     .withDescription(
-                            "The interval before a savepoint trigger attempt is marked as unsuccessful");
+                            "The interval before a savepoint trigger attempt is marked as unsuccessful.");
```
Comment (SteNicholas) [inline flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/config/OperatorConfigOptions.java:64]: Ditto.

File: docs/content/docs/deployment/configuration.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
+---
```
Comment (SteNicholas) [inline docs/content/docs/deployment/configuration.md:1]: Could the document of configuration be generated by `OperatorConfigOptions`?

Comment (gyfora) [reply docs/content/docs/deployment/configuration.md:1]: Not very easy to do unfortunately. flink-docs contains the annotaiton processing logic for flink but it's not flexible

**Review by gyfora — CHANGES_REQUESTED** [review]
The changes look good but I would prefer to have them in the `operations` section for now.

File: flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/config/OperatorConfigOptions.java (hunk 1 above)
Comment (bgeng777) [reply flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/config/OperatorConfigOptions.java:36]: Thanks for the information. As we do not automatically generate documents for now, I will roll back this change and only keep it in our doc website.

**Comment by bgeng777** [comment]
> The changes look good but I would prefer to have them in the `operations` section for now.

I am ok to both. Previous design follows the pattern of flink doc. Now, only a `configuration` page is added in `operations` sec.

**Comment by gyfora** [comment]
Merging this, @bgeng777 please open a backport PR for the `release-0.1` branch

**Pull request merged**